BH/Drawing/Stats/StatsDisplay.cpp

```diff
@@ -1,505 +1,938 @@
-#include "../../D2Ptrs.h"
+
 #include "../../BH.h"
 #include "../../D2Helpers.h"
+#include "../../D2Ptrs.h"
+
 #include "../../Modules/Item/Item.h"
 
+#include "../Basic/Boxhook/Boxhook.h"
+#include "../Basic/Framehook/Framehook.h"
+#include "../Basic/Texthook/Texthook.h"
+
+
 using namespace Drawing;
 
-std::vector<int> FHRBreakpoints[] = {
-	{6 , 13 , 20 , 32 , 52 , 86 , 174 , 600},
-	{5 , 9 , 14 , 20 , 30 , 42 , 60 , 86 , 142 , 280},
-	{5 , 10 , 16 , 26 , 39 , 56 , 86 , 152 , 377},
-	{7 , 15 , 27 , 48 , 86 , 200 , 4680},
-	{7 , 15 , 27 , 48 , 86 , 200 , 4680},
-	{5 , 10 , 16 , 26 , 39 , 56 , 86 , 152 , 377},
-	{7 , 15 , 27 , 48 , 86 , 200 , 4680}
+const int kWolfTxtFileNoAlias         = 139;
+const int kBearTxtFileNoAlias         = 140;
+const int kDruidOneHandFHRAlias       = 141;
+const int kPaladinSpearStaveFHRAlias  = 142;
+const int kSorceressLightningFCRAlias = 143;
+
+std::map<DWORD, std::vector<int>> faster_hit_recovery_frames = {
+    { CLASS_AMA, { 6, 13, 20, 32, 52, 86, 174, 600 } },
+    { CLASS_SOR, { 5, 9, 14, 20, 30, 42, 60, 86, 142, 280 } },
+    { CLASS_NEC, { 5, 10, 16, 26, 39, 56, 86, 152, 377 } },
+    { CLASS_DRU, { 5, 10, 16, 26, 39, 56, 86, 152, 377 } },
+    { CLASS_BAR, { 7, 15, 27, 48, 86, 200 } },
+    { CLASS_ASN, { 7, 15, 27, 48, 86, 200 } },
+    { CLASS_PAL, { 7, 15, 27, 48, 86, 200 } },
+    { kWolfTxtFileNoAlias, { 9, 20, 42, 86, 280 } },
+    { kBearTxtFileNoAlias, { 5, 10, 16, 24, 37, 54, 86, 152, 360 } },
+    { kDruidOneHandFHRAlias, { 3, 7, 13, 19, 29, 42, 63, 99, 174, 456 } },
+    { kPaladinSpearStaveFHRAlias, { 3, 7, 13, 20, 32, 48, 75, 129, 280 } },
+    { MERC_A1, { 6, 13, 20, 32, 52, 86, 174, 600 } },
+    { MERC_A2, { 5, 9, 14, 20, 30, 42, 60, 86, 142, 280 } },
+    { MERC_A3, { 5, 8, 13, 18, 24, 32, 46, 63, 86, 133, 232, 600 } },
+    { MERC_A5, { 7, 15, 27, 48, 86, 200 } },
 };
 
-std::vector<int> FCRBreakpoints[] = {
-	{7 , 14 , 22 , 32 , 48 , 68 , 99 , 152},
-	{9 , 20 , 37 , 63 , 105 , 200},
-	{9 , 18 , 30 , 48 , 75 , 125},
-	{9 , 18 , 30 , 48 , 75 , 125},
-	{9 , 20 , 37 , 63 , 105 , 200},
-	{4 , 10 , 19 , 30 , 46 , 68 , 99 , 163},
-	{8 , 16 , 27 , 42 , 65 , 102 , 174}
+std::map<DWORD, std::vector<int>> faster_cast_rate_frames = {
+    { CLASS_AMA, { 7, 14, 22, 32, 48, 68, 99, 152 } },
+    { CLASS_SOR, { 9, 20, 37, 63, 105, 200 } },
+    { CLASS_BAR, { 9, 20, 37, 63, 105, 200 } },
+    { CLASS_NEC, { 9, 18, 30, 48, 75, 125 } },
+    { CLASS_PAL, { 9, 18, 30, 48, 75, 125 } },
+    { CLASS_ASN, { 8, 16, 27, 42, 65, 102, 174 } },
+    { CLASS_DRU, { 4, 10, 19, 30, 46, 68, 99, 163 } },
+    { kWolfTxtFileNoAlias, { 6, 14, 26, 40, 60, 95, 157 } },
+    { kBearTxtFileNoAlias, { 7, 15, 26, 40, 63, 99, 163 } },
+    { kSorceressLightningFCRAlias, { 7, 15, 23, 35, 52, 78, 117, 194 } },
+    { MERC_A1, { 0 } },
+    { MERC_A2, { 0 } },
+    { MERC_A3, { 8, 15, 26, 39, 58, 86, 138 } },
+    { MERC_A5, { 0 } },
+};
+
+std::map<DWORD, std::vector<int>> increased_attack_speed_frames = {
```

Comment:
I don't quite understand what this is, just a static number based on character class? If it's unfinished, we should remove the display in the advanced stat screen. If we keep it, it needs to be renamed from "IAS" to something else

Reply:
This stub was put in to handle IAS breakpoints for the act 2 mercenary originally, and was going to expand to each character class. I still have yet to find a comprehensive list of min frames/max frames breakpoints for every skill. In the future the static number could be removed in favor of breakpoints in the same way that FCR changes based on the ability.